redis/src/aio/connection.rs

```diff
@@ -361,13 +361,18 @@ where
     }
 }
 
-async fn get_socket_addrs(host: &str, port: u16) -> RedisResult<SocketAddr> {
+async fn get_socket_addrs(
+    host: &str,
+    port: u16,
+) -> RedisResult<impl Iterator<Item = SocketAddr> + Send + '_> {
     #[cfg(feature = "tokio-comp")]
-    let mut socket_addrs = lookup_host((host, port)).await?;
+    let socket_addrs = lookup_host((host, port)).await?;
     #[cfg(all(not(feature = "tokio-comp"), feature = "async-std-comp"))]
     let mut socket_addrs = (host, port).to_socket_addrs().await?;
```

Comment:
if you didn't need `mut` for tokio, you probably don't need it for std-async either.